multibody/math/spatial_acceleration.h

```diff
@@ -159,8 +159,9 @@ class SpatialAcceleration : public SpatialVector<SpatialAcceleration, T> {
   SpatialAcceleration<T> Shift(
       const Vector3<T>& offset,
       const Vector3<T>& angular_velocity_of_this_frame) const {
-    return SpatialAcceleration<T>(*this).ShiftInPlace(
-        offset, angular_velocity_of_this_frame);
+    SpatialAcceleration<T> result(*this);
+    result.ShiftInPlace(offset, angular_velocity_of_this_frame);
+    return result;
```

Comment:
In the other PR, you recommended that we need a clear comment so that nobody reverts the change. If there is no test coverage, presumably we need that here?